### PR TITLE
8342508: Use latch in BasicMenuUI/bug4983388.java instead of delay

### PR DESCRIPTION
**Pull request by aivanov-jdk**
Use a `CountDownLatch` in `javax/swing/plaf/basic/BasicMenuUI/4983388/bug4983388.java` instead of delay.

The latch provides a direct way to synchronise EDT and main thread, the test will finish quicker.

I ran the updated test a few times with `JTREG=REPEAT_COUNT=20`, the test always passed in the CI.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8342508](https://bugs.openjdk.org/browse/JDK-8342508): Use latch in BasicMenuUI/bug4983388.java instead of delay (**Bug** - P5)

### Reviewers
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**) Review applies to [10b51b09](https://git.openjdk.org/jdk/pull/21585/files/10b51b09284a3bde942eb04c0f444bad17cacf02)
 * @abdelhak-zaaim (no known openjdk.org user name / role) Review applies to [10b51b09](https://git.openjdk.org/jdk/pull/21585/files/10b51b09284a3bde942eb04c0f444bad17cacf02)
 * [Alexander Zvegintsev](https://openjdk.org/census#azvegint) (@azvegint - **Reviewer**)
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21585/head:pull/21585` \
`$ git checkout pull/21585`

Update a local copy of the PR: \
`$ git checkout pull/21585` \
`$ git pull https://git.openjdk.org/jdk.git pull/21585/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21585`

View PR using the GUI difftool: \
`$ git pr show -t 21585`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21585.diff">https://git.openjdk.org/jdk/pull/21585.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21585#issuecomment-2422710065)

### CODE REVIEW

**Review by mrserb — APPROVED** [review]

**Review by abdelhak-zaaim — APPROVED** [review]

**Review by azvegint — APPROVED** [review]

File: test/jdk/javax/swing/plaf/basic/BasicMenuUI/4983388/bug4983388.java [hunk 1]
```diff
@@ -56,21 +69,21 @@ private static void createAndShowGUI() {
...
-
         try {
             UIManager.setLookAndFeel("com.sun.java.swing.plaf.gtk.GTKLookAndFeel");
         } catch (UnsupportedLookAndFeelException | ClassNotFoundException ex) {
-            System.err.println("GTKLookAndFeel is not supported on this platform. Using defailt LaF for this platform.");
+            System.err.println("GTKLookAndFeel is not supported on this platform. "
+                               + "Using default LaF for this platform.");
         }
 
         SwingUtilities.invokeAndWait(new Runnable() {
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/plaf/basic/BasicMenuUI/4983388/bug4983388.java:89]: `createAndShowGUI` method is `static`, I suggest using method reference.

**Review by kumarabhi006 — APPROVED** [review]
Looks ok to me except for a suggestion.

File: test/jdk/javax/swing/plaf/basic/BasicMenuUI/4983388/bug4983388.java (hunk 1 above)
Comment (aivanov-jdk) [reply test/jdk/javax/swing/plaf/basic/BasicMenuUI/4983388/bug4983388.java]: Method references can be used in non-static context by providing an instance to the left of the double colon.

I did it on Friday but I forgot to push the change.

**Review by kumarabhi006 — APPROVED** [review]

**Pull request closed without merging**